File: ios/Zedu/CallKeepAwake.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(CallKeepAwake, NSObject)

RCT_EXTERN_METHOD(setEnabled:(BOOL)enabled
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

@end

File: ios/Zedu/CallKeepAwake.swift
import Foundation
import UIKit
import React

@objc(CallKeepAwake)
class CallKeepAwake: NSObject {
  @objc
  static func requiresMainQueueSetup() -> Bool {
    true
  }

  @objc
  func setEnabled(_ enabled: Bool,
                  resolver resolve: @escaping RCTPromiseResolveBlock,
                  rejecter reject: @escaping RCTPromiseRejectBlock) {
    DispatchQueue.main.async {
      UIApplication.shared.isIdleTimerDisabled = enabled
      resolve(nil)
    }
  }
}
